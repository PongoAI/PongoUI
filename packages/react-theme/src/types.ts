import type { ColorSet } from '@pongo-ui/react-theme-generator';

export type Theme = ColorTokens & Fonts;

/**
 * Color theme tokens.
 */
export type ColorTokens = { textColor: string; canvasColor: string } & ColorSet<
  'inherit' | 'brand' | 'secondary' | 'success' | 'danger' | 'social' | 'warning' | 'info'
>;

export type Fonts = FontFamilyTokens & FontOpacityTokens & FontSizeTokens & FontLineHeightTokens & FontWeightTokens;

/**
 * Font family tokens.
 */
export type FontFamilyTokens = {
  baseFont: string;
  monospaceFont: string;
};

/**
 * Font opacity tokens
 */
export type FontOpacityTokens = {
  display: string;
  subHeadline: string;
  body: string;
  caption: string;
  title: string;
  header1: string;
  header2: string;
  header3: string;
};

/**
 * Font size tokens.
 */
export type FontSizeTokens = {
  fontSize100: string;
  fontSize200: string;
  fontSize300: string;
  fontSize400: string;
  fontSize500: string;
  fontSize600: string;
  fontSize700: string;
  fontSize800: string;
  fontSize900: string;
  fontSize1000: string;
};

/**
 * Font line height tokens.
 */
export type FontLineHeightTokens = {
  lineHeight100: string;
  lineHeight200: string;
  lineHeight300: string;
  lineHeight400: string;
  lineHeight500: string;
  lineHeight600: string;
  lineHeight700: string;
  lineHeight800: string;
  lineHeight900: string;
  lineHeight1000: string;
};

/**
 * Font weight tokens.
 */
export type FontWeightTokens = {
  thin: number;
  light: number;
  regular: number;
  medium: number;
  bold: number;
  black: number;
};
