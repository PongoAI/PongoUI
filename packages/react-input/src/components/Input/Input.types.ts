import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type InputSlots = {
  /**
   * The root of the Input.
   */
  root: Slot<'span'>;

  /**
   * The inner nearest content to render within the Input.
   */
  contentBefore?: Slot<'span'>;

  /**
   * The inner furthest content to render within the Input.
   */
  contentAfter?: Slot<'span'>;

  /**
   * Hidden input used to handle the Input's functionality.
   */
  input?: Slot<'input'>;
};

type InputCommons = {
  /**
   * The starting value for a uncontrolled Input. If `true` then the Input will be enabled.
   * Mutually exclusive with `value` prop.
   *
   * @defaultValue ''
   */
  defaultValue?: string;

  /**
   * The current value for a controlled Input. If `true` then the Input will be enabled.
   * Mutually exclusive with `defaultValue` prop.
   */
  value?: string;

  /**
   * The autocomplete hint type for the Input.
   */
  autocomplete?: string;

  /**
   * The label that appears in the Input when no value is set. When a value is set the label will not be visible.
   * Mutually exclusive with `label` prop.
   */
  placeholder?: string;

  /**
   * Whether the Input should be disabled.
   *
   * @defaultValue false
   */
  disabled?: boolean;

  /**
   * Whether the Input should be in an error state.
   *
   * @defaultValue false
   */
  error?: boolean;

  /**
   * Whether the Input should be required.
   *
   * @defaultValue false
   */
  required?: boolean;

  /**
   * The appearance variant to use for the Input.
   *
   * @defaultValue outline
   */
  appearance?: 'outline' | 'standard' | 'filled';

  /**
   * Callback to be called when the value changes.
   */
  onChange?: (
    ev: React.ChangeEvent<HTMLInputElement>,
    data: {
      value: string;
    },
  ) => void;

  /**
   * The size of the Input control.
   *
   * @default medium
   */
  size?: 'small' | 'medium' | 'large';

  /**
   * The type of content for the the Input element.
   */
  type?:
    | 'text'
    | 'email'
    | 'password'
    | 'search'
    | 'tel'
    | 'url'
    | 'date'
    | 'datetime-local'
    | 'month'
    | 'number'
    | 'time'
    | 'week';
};

export type InputProps = ComponentProps<InputSlots> & InputCommons;

export type InputState = ComponentState<InputSlots> & InputCommons;
