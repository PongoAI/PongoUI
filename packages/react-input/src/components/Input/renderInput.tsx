import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { InputSlots, InputState } from './Input.types';

export const renderInput = (state: InputState) => {
  const { slots, slotProps } = getSlots<InputSlots>(state);

  return (
    <slots.root {...slotProps.root}>
      <slots.input {...slotProps.input} />
    </slots.root>
  );
};
