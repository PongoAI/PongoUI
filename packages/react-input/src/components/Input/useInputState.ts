import * as React from 'react';
import { useControllableState, useEventCallback, useId } from '@fluentui/react-utilities';
import type { InputState } from './Input.types';

export const useInputState = (state: InputState) => {
  const {
    label,
    appearance,
    defaultValue,
    value,
    size,
    danger,
    helperText,
    contentBefore,
    contentAfter,
    disabled,
    onChange,
  } = state;

  const { id } = state.input;
  const labelId = label ? useId('input-label', id) : undefined;
  const helperTextId = helperText ? useId('input-helper-text-', id) : undefined;

  const [currentValue, setCurrentValue] = useControllableState({
    defaultState: defaultValue,
    state: value,
    initialState: '',
  });

  /**
   * Updates the controlled `currentValue` to the new `incomingValue` and clamps it.
   */
  const onInputChange = useEventCallback((ev: React.ChangeEvent<HTMLInputElement>) => {
    const newValue = ev.target.value;
    onChange?.(ev, { value: newValue });
    setCurrentValue(newValue);
  });

  // Root Border Wrapper Props
  state.root.appearance = appearance;
  state.root.danger = danger;
  state.root.disabled = disabled;
  state.root.size = size;
  state.root.contentBefore = contentBefore;
  state.root.contentAfter = contentAfter;
  state.root.currentValue = currentValue;

  if (helperText) {
    state.root.helperText = helperText;
    state.root.helperTextId = helperTextId;
    state.input['aria-describedby'] = helperTextId;
  }

  if (label) {
    state.root.label = label;
    state.root.labelId = labelId;
    state.input.id = labelId;
  }

  // Input Element props
  state.input.value = currentValue;
  state.input.onInput = onInputChange;
  state.input['aria-invalid'] = danger;

  return state;
};
