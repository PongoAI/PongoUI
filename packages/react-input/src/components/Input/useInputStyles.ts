import { shorthands, makeStyles, mergeClasses } from '@griffel/react';
import { createCustomFocusIndicatorStyle } from '@fluentui/react-tabster';
import type { InputState } from './Input.types';
import { tokens } from '@pongo-ui/react-theme';

export const inputHeight = `--pongoai-input-height`;

export const useStyles = makeStyles({
  root: {
    position: 'relative',
    height: inputHeight,
    boxSizing: 'border-box',
    display: 'flex',
    alignItems: 'center',
    flexWrap: 'nowrap',
    color: tokens.textColor,
    backgroundColor: tokens.canvasColor,
    fontFamily: tokens.baseFont,
  },

  // Appearance
  outline: {
    border: `2px solid ${tokens.inherit}`,
    borderRadius: '4px',
  },

  standard: {
    borderBottom: `2px solid ${tokens.inherit}`,
  },

  filled: {
    background: tokens.inheritForegroundPressed,
    ...shorthands.borderRadius(tokens.rounded, tokens.rounded, tokens.square, tokens.square),
    ...shorthands.borderBottom(`2px solid ${tokens.inherit}`),
  },

  // Size
  small: {
    height: '35px',
    width: '150px',
    fontSize: tokens.fontSize200,
  },

  medium: {
    height: '45px',
    width: '200px',
    fontSize: tokens.fontSize300,
  },

  large: {
    height: '50px',
    width: '300px',
    fontSize: tokens.fontSize400,
  },

  // Focus
  focusIndicator: createCustomFocusIndicatorStyle(
    {
      ':after': {
        content: "''",
        position: 'absolute',
        top: '-8px',
        right: '-8px',
        bottom: '-8px',
        left: '-8px',
        boxSizing: 'border-box',
        ...shorthands.border('2px', 'solid', tokens.textColor),
        ...shorthands.borderRadius(tokens.rounded),
      },
    },
    { selector: 'focus-within' },
  ),
});

const useContentStyles = makeStyles({
  base: {
    boxSizing: 'border-box',
    '> svg': { display: 'block' },
  },

  enabled: {
    color: tokens.inherit,
    fill: tokens.inherit,
  },

  disabled: {
    color: tokens.inheritDisabled,
    fill: tokens.inheritDisabled,
  },
});

export const useInputElementStyles = makeStyles({
  input: {
    ...shorthands.margin('0px'),
    ...shorthands.padding('0px'),
    ...shorthands.borderStyle('none'),
    boxSizing: 'border-box',
    flexGrow: 1,
    minWidth: 0,
    height: '100%',
    paddingLeft: '10px',
    paddingRight: '10px',
    backgroundColor: 'red',
    ':focus-visible': {
      outlineStyle: 'none',
    },
  },

  enabled: {
    color: tokens.textColor,
    '::placeholder': {
      color: tokens.canvasColor,
      opacity: 1,
    },
  },

  disabled: {
    color: tokens.inheritDisabled,
    '::placeholder': {
      color: tokens.inheritDisabled,
      opacity: 1,
    },
  },
});

export const useInputStyles = (state: InputState) => {
  const styles = useStyles();
  const inputStyles = useInputElementStyles();
  const contentStyles = useContentStyles();

  state.root.className = mergeClasses(
    styles.root,
    styles[state.size!],
    styles[state.appearance!],
    styles.focusIndicator,
    state.root.className,
  );

  state.input.className = mergeClasses(
    inputStyles.input,
    state.disabled ? inputStyles.disabled : inputStyles.enabled,
    state.input.className,
  );

  const contentClasses = [contentStyles.base, state.disabled ? contentStyles.disabled : contentStyles.enabled];

  if (state.contentBefore) {
    state.contentBefore.className = mergeClasses(...contentClasses, state.contentBefore.className);
  }
  if (state.contentAfter) {
    state.contentAfter.className = mergeClasses(...contentClasses, state.contentAfter.className);
  }

  return state;
};
