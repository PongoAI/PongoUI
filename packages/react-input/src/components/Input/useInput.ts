import * as React from 'react';
import { getPartitionedNativeProps, resolveShorthand } from '@fluentui/react-utilities';
import { useInputState } from './useInputState';
import type { InputProps, InputState } from './Input.types';
import { InputWrapper } from '../InputWrapper';

export const useInput = (props: InputProps, ref: React.Ref<HTMLInputElement>): InputState => {
  const {
    value,
    defaultValue,
    onChange,
    appearance = 'outline',
    contentBefore,
    contentAfter,
    size = 'medium',
    disabled,
    danger,
  } = props;

  const nativeProps = getPartitionedNativeProps({
    props,
    primarySlotTagName: 'input',
    excludedPropNames: ['size', 'onChange', 'value', 'defaultValue'],
  });

  const state: InputState = {
    value,
    defaultValue,
    onChange,
    size,
    appearance,
    disabled,
    danger,
    contentBefore,
    contentAfter,
    components: {
      root: InputWrapper,
      input: 'input',
    },
    root: resolveShorthand(props.root, {
      required: true,
      defaultProps: nativeProps.root,
    }),
    input: resolveShorthand(props.input, {
      required: true,
      defaultProps: {
        type: 'text',
        ref,
        ...nativeProps.primary,
      },
    }),
  };

  useInputState(state);

  return state;
};
