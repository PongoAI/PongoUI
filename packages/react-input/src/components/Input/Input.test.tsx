import React from 'react';
import { Input } from './Input';
import { InputProps } from './Input.types';
import { isConformant } from '../../common/isConformant';

describe('Input', () => {
  isConformant({
    Component: Input as React.FunctionComponent<InputProps>,
    displayName: 'Input',
  });
});
