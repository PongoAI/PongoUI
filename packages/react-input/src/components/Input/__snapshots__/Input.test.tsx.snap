// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders a basic input 1`] = `
<body>
  <div>
    <span
      class="pongo-input-wrapper ___y0dmvi0 f10pi13n f1ewtqcl ftuwxu6 fv68sux fp6vxd fa4t5tb flvyvdh fc467gg f8qjfyk fk6fouc f4d9j23 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp fifp7yv f1ov4xf1 f1asdtw4 f1ern45e f1n71otn f1h8hb77 f1deefiw f1arxnym fqzc6w8 fuzzvh5 fxf9f1y frdto5n ff0gctb f3zr33r f8g0ba1 f5uakvk frcipv9 f199l62q f1pa2qi2 fyqop9u f1ooiv2t f1bwkb58 fiio4mc f1m5ya7j fc8z0qo fmw406p f12c27rx f17thq47 f12kdog7 f5qrn4q f1kndqco f1kc1xys f1d7l9b4 f16n8puj fkcy365 f5le7dr f170vdtw fg4eyww f1cngt4b f17x5ctf fbk6y72 f1uajmgh f11t0hz4 f2v3se5 f18saz9o f80sxui f19q3xa0 fzmdvw2 fbh30h8"
      data-tabster="{\\"groupper\\":{\\"tabbability\\":2}}"
      style="--pongo-input-line-height: 45px; --pongo-input-width: 200px;"
    >
      <input
        class="___19ah7w0 f1klrjes fsyuiy0 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp f81rol6 f1ov4xf1 frdkuqy f1ern45e f1n71otn f1h8hb77 f1deefiw f1rbvai9 f13ju9t4 f1lh17z1 f1dlw1sp f1ewtqcl fqerorx f1l02sjl f3rmtva f2hkw1w f113hnb5 fc467gg f1vw2ftv f13ta7ih f1arxnym"
        type="text"
        value=""
      />
    </span>
  </div>
</body>
`;

exports[`Input renders a danger input 1`] = `
<body>
  <div>
    <span
      class="pongo-input-wrapper ___vt1aw30 f10pi13n f1ewtqcl ftuwxu6 fv68sux fp6vxd fa4t5tb flvyvdh fc467gg f8qjfyk fk6fouc f4d9j23 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp fifp7yv f1ov4xf1 f1asdtw4 f1ern45e f1n71otn f1h8hb77 f1deefiw f1arxnym fqzc6w8 fuzzvh5 fxf9f1y frdto5n ff0gctb f3zr33r f8g0ba1 f5uakvk frcipv9 f199l62q f1pa2qi2 fyqop9u f1ooiv2t f1bwkb58 fiio4mc f1m5ya7j fc8z0qo f1tsuuen f1jyinwm f1gmfdaf ff4t21v f5qrn4q f1kndqco f1kc1xys f1d7l9b4 f16n8puj fkcy365 f5le7dr f170vdtw f1md1kfu f1385xuj fwuljub fophbcs f1uajmgh f11t0hz4 f2v3se5 f18saz9o f4heoct fzmdvw2 fbh30h8"
      data-tabster="{\\"groupper\\":{\\"tabbability\\":2}}"
      style="--pongo-input-line-height: 45px; --pongo-input-width: 200px;"
    >
      <input
        aria-invalid="true"
        class="___a90x680 f1klrjes fsyuiy0 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp f81rol6 f1ov4xf1 frdkuqy f1ern45e f1n71otn f1h8hb77 f1deefiw f1rbvai9 f13ju9t4 f1lh17z1 f1dlw1sp f1ewtqcl fqerorx f1l02sjl f3rmtva f2hkw1w f113hnb5 fc467gg f1lxi41g f13ta7ih f1arxnym"
        type="text"
        value=""
      />
    </span>
  </div>
</body>
`;

exports[`Input renders a disabled input 1`] = `
<body>
  <div>
    <span
      class="pongo-input-wrapper ___136cyr8 f10pi13n f1ewtqcl ftuwxu6 fv68sux fp6vxd fa4t5tb flvyvdh fc467gg f8qjfyk fk6fouc f4d9j23 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp fifp7yv f1ov4xf1 f1asdtw4 f1ern45e f1n71otn f1h8hb77 f1deefiw f1arxnym fqzc6w8 fuzzvh5 fxf9f1y frdto5n ff0gctb f3zr33r f8g0ba1 f5uakvk frcipv9 f199l62q f1pa2qi2 fyqop9u f1ooiv2t f1bwkb58 fiio4mc f1m5ya7j fc8z0qo flm3n1z f1pltsua f15x2d8x fhngjcl f5qrn4q f1kndqco f1kc1xys f1d7l9b4"
      data-tabster="{\\"groupper\\":{\\"tabbability\\":2}}"
      style="--pongo-input-line-height: 45px; --pongo-input-width: 200px;"
    >
      <input
        class="___ll66jt0 f1klrjes fsyuiy0 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp f81rol6 f1ov4xf1 frdkuqy f1ern45e f1n71otn f1h8hb77 f1deefiw f1rbvai9 f13ju9t4 f1lh17z1 f1dlw1sp f1ewtqcl fqerorx f1l02sjl f3rmtva f2hkw1w fdrzuqr favx1wl fgbidd3 f13ta7ih f1arxnym"
        disabled=""
        type="text"
        value=""
      />
    </span>
  </div>
</body>
`;

exports[`Input renders a label Input 1`] = `
<body>
  <div>
    <span
      class="pongo-input-wrapper ___y0dmvi0 f10pi13n f1ewtqcl ftuwxu6 fv68sux fp6vxd fa4t5tb flvyvdh fc467gg f8qjfyk fk6fouc f4d9j23 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp fifp7yv f1ov4xf1 f1asdtw4 f1ern45e f1n71otn f1h8hb77 f1deefiw f1arxnym fqzc6w8 fuzzvh5 fxf9f1y frdto5n ff0gctb f3zr33r f8g0ba1 f5uakvk frcipv9 f199l62q f1pa2qi2 fyqop9u f1ooiv2t f1bwkb58 fiio4mc f1m5ya7j fc8z0qo fmw406p f12c27rx f17thq47 f12kdog7 f5qrn4q f1kndqco f1kc1xys f1d7l9b4 f16n8puj fkcy365 f5le7dr f170vdtw fg4eyww f1cngt4b f17x5ctf fbk6y72 f1uajmgh f11t0hz4 f2v3se5 f18saz9o f80sxui f19q3xa0 fzmdvw2 fbh30h8"
      data-tabster="{\\"groupper\\":{\\"tabbability\\":2}}"
      style="--pongo-input-line-height: 45px; --pongo-input-width: 200px;"
    >
      <label
        class="pongo-input-label ___3vyzt00 f1euv43f f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp f12huiiw f1ov4xf1 f161knb0 fpnytvn f9j3dip f1ysjbbf fk6fouc fz5stix f1q148li f1cmbuwj f1sdsnyy f1aehjj5 f1xqy1su ftuwxu6 fi64zpg fc7pyfw f122n59 fvz7k9s fkfijr0 fsxfru3 fuzzvh5 fxf9f1y ff0gctb f8g0ba1 fsa65no f5uakvk frcipv9 fsm07gi f1qo5dib fb322o2"
        for="input-label1"
      >
        test
      </label>
      <input
        class="___19ah7w0 f1klrjes fsyuiy0 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp f81rol6 f1ov4xf1 frdkuqy f1ern45e f1n71otn f1h8hb77 f1deefiw f1rbvai9 f13ju9t4 f1lh17z1 f1dlw1sp f1ewtqcl fqerorx f1l02sjl f3rmtva f2hkw1w f113hnb5 fc467gg f1vw2ftv f13ta7ih f1arxnym"
        id="input-label1"
        type="text"
        value=""
      />
    </span>
  </div>
</body>
`;
